python3-sys/src/pylifecycle.rs

```diff
@@ -0,0 +1,50 @@
+// There are 2 pylifecycle.h in CPython. We currently only define the Py_LIMITED_API
+// symbols because other symbols exist in their legacy locations in other modules.
+
+#[cfg(Py_3_8)]
+#[cfg(not(Py_LIMITED_API))]
+use libc::{c_char, c_int, c_void, wchar_t, FILE};
+#[cfg(Py_3_8)]
+#[cfg(not(Py_LIMITED_API))]
+use crate::initconfig::{PyPreConfig, PyStatus, PyConfig};
+#[cfg(Py_3_8)]
+#[cfg(not(Py_LIMITED_API))]
+use crate::object::PyObject;
+#[cfg(Py_3_8)]
+#[cfg(not(Py_LIMITED_API))]
+use crate::pyport::Py_ssize_t;
+
+// Symbols from Include/pylifecycle.h
+
+// TODO move these symbols from their legacy locations into this module.
+
+// Symbols from Include/cpython/pylifecycle.h
+
+#[cfg(Py_3_8)]
+#[cfg(not(Py_LIMITED_API))]
+#[cfg_attr(windows, link(name = "pythonXY"))]
+extern "C" {
+    pub fn Py_PreInitialize(src_config: *const PyPreConfig) -> PyStatus;
+    pub fn Py_PreInitializeFromBytesArgs(src_config: *const PyPreConfig, argc: Py_ssize_t, argv: *mut *mut c_char) -> PyStatus;
+    pub fn Py_PreInitializeFromArgs(src_config: *const PyPreConfig, argc: Py_ssize_t, argv: *mut *mut wchar_t) -> PyStatus;
+
+    pub fn _Py_IsCoreInitialized() -> c_int;
+
+    pub fn Py_InitializeFromConfig(config: *const PyConfig) -> PyStatus;
+    pub fn _Py_InitializeMain() -> PyStatus;
```

Comment:
(detail) I suppose you omitted `_Py_InitializeFromArgs` and `_Py_InitializeFromWideArgs` because they have been removed for the future Python 3.9 (that's at least what I see in 3.9.0a4). Then a comment about that would be nice.

Reply:
More or less. They were removed from 3.9. But the reason they were removed was because they were declared in the `.h` but had no implementation! These symbols caused linking failures on CI due to missing symbols!